File: back-end/aths/src/utils/logger.js
/**
 * Logger Utility
 * Centralized logging for the application
 */

const LOG_LEVEL = process.env.LOG_LEVEL || 'info';
const NODE_ENV = process.env.NODE_ENV || 'development';

// Log levels
const levels = {
  error: 0,
  warn: 1,
  info: 2,
  debug: 3,
};

const currentLevel = levels[LOG_LEVEL] || levels.info;

/**
 * Format log message with timestamp
 */
const formatMessage = (level, message, meta = {}) => {
  const timestamp = new Date().toISOString();
  const metaString = Object.keys(meta).length > 0 ? JSON.stringify(meta) : '';
  return `[${timestamp}] [${level.toUpperCase()}] ${message} ${metaString}`.trim();
};

/**
 * Log error messages
 */
export const error = (message, meta = {}) => {
  if (currentLevel >= levels.error) {
    console.error(formatMessage('error', message, meta));
  }
};

/**
 * Log warning messages
 */
export const warn = (message, meta = {}) => {
  if (currentLevel >= levels.warn) {
    console.warn(formatMessage('warn', message, meta));
  }
};

/**
 * Log info messages
 */
export const info = (message, meta = {}) => {
  if (currentLevel >= levels.info) {
    console.log(formatMessage('info', message, meta));
  }
};

/**
 * Log debug messages
 */
export const debug = (message, meta = {}) => {
  if (currentLevel >= levels.debug && NODE_ENV === 'development') {
    console.log(formatMessage('debug', message, meta));
  }
};

/**
 * Log HTTP requests
 */
export const logRequest = (req) => {
  info(`${req.method} ${req.originalUrl}`, {
    ip: req.ip,
    userAgent: req.get('user-agent'),
  });
};

/**
 * Log authentication events
 */
export const logAuth = (event, userId, meta = {}) => {
  info(`Auth Event: ${event}`, {
    userId,
    ...meta,
  });
};

/**
 * Log security events
 */
export const logSecurity = (event, meta = {}) => {
  warn(`Security Event: ${event}`, meta);
};

export default {
  error,
  warn,
  info,
  debug,
  logRequest,
  logAuth,
  logSecurity,
};

File: back-end/cmps/app/utils/logger.py
"""Logging utility for the application"""
import logging

# Import settings with try/except to handle initialization issues
try:
    from app.config.settings import settings
    SERVICE_NAME = settings.SERVICE_NAME
    LOG_LEVEL = settings.LOG_LEVEL
except Exception:
    SERVICE_NAME = "complaint-service"
    LOG_LEVEL = "INFO"

# Configure logging
logging.basicConfig(
    level=getattr(logging, LOG_LEVEL, logging.INFO),
    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
    datefmt='%Y-%m-%d %H:%M:%S'
)

logger = logging.getLogger(SERVICE_NAME)


def info(message: str):
    """Log info message"""
    logger.info(message)


def error(message: str):
    """Log error message"""
    logger.error(message)


def warning(message: str):
    """Log warning message"""
    logger.warning(message)


def debug(message: str):
    """Log debug message"""
    logger.debug(message)
